### PR TITLE
[ASM] - Expander - Enrichment Status and N/A values (3581 & 4820)

### PR DESCRIPTION
**Pull request by BigEasyJ**
Please ignore #28573 and only refer to this PR.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## NOTE

This PR has a release version higher than #28663. If merged before #28375 please change the version.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
- [EXPANDR-4820](https://jira-hq.paloaltonetworks.local/browse/EXPANDR-4820)
- [EXPANDR-3851](https://jira-hq.paloaltonetworks.local/browse/EXPANDR-3851)

## Description
- Added a new field, **ASM - Enrichment Status**, to track the status of types of platforms enrichment has been performed with.
- Updated all sub-playbooks in `Cortex ASM - Enrichment` to incorporate the `ASM - Enrichment Status` field to identify if enrichment was completed successfully or not.

## Must have
- [x] Tests (internal)
- [x] Documentation (internal)

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , can you please look over?

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 1]
```diff
@@ -950,17 +955,644 @@ tasks:
...
+      name: Set false flag for completed enrichment
+      description: Set a value in context under the key you entered.
+      scriptName: Set
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "20"
+    scriptarguments:
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1013]:
```suggestion
    scriptarguments:
      append:
        simple: "true"
```
Seem to always append in other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 2]
```diff
@@ -950,17 +955,644 @@ tasks:
...
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1128]:
```suggestion
    scriptarguments:
      append:
        simple: "true"
```
Seem to always append in other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 3]
```diff
@@ -950,17 +955,644 @@ tasks:
...
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1243]:
```suggestion
    scriptarguments:
      append:
        simple: "true"
```
Seem to always append in other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Prisma_Cloud_Enrichment.yml [hunk 4]
```diff
@@ -383,8 +355,208 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "10":
+    id: "10"
+    taskid: cf6daeee-77c3-495d-870e-e0d15087e37a
+    type: regular
+    task:
+      id: cf6daeee-77c3-495d-870e-e0d15087e37a
+      version: -1
+      name: Set flag for completed enrichment
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Prisma_Cloud_Enrichment.yml:376]:
```suggestion
      name: Set true flag for completed enrichment
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 5]
```diff
@@ -950,17 +955,644 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "43":
+    id: "43"
+    taskid: 7a4df59a-49f0-4182-8add-7e5c1f352bbd
+    type: regular
+    task:
+      id: 7a4df59a-49f0-4182-8add-7e5c1f352bbd
+      version: -1
+      name: Set ASM enrichment status field (s3)
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1271]:
```suggestion
      name: Set ASM enrichment status to true (s3)
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 6]
```diff
@@ -950,17 +955,644 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "54":
+    id: "54"
+    taskid: 27904728-1475-4457-8bac-23ceefd53134
+    type: regular
+    task:
+      id: 27904728-1475-4457-8bac-23ceefd53134
+      version: -1
+      name: Set ASM enrichment status field (s3)
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1518]:
```suggestion
      name: Set ASM enrichment status to false (s3)
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 7]
```diff
@@ -950,17 +955,644 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "44":
+    id: "44"
+    taskid: 77d86402-7d2f-4fa3-8342-f6b81ec9e377
+    type: regular
+    task:
+      id: 77d86402-7d2f-4fa3-8342-f6b81ec9e377
+      version: -1
+      name: Set ASM enrichment status field (AWS)
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1316]:
```suggestion
      name: Set ASM enrichment status to true (AWS)
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml [hunk 8]
```diff
@@ -950,17 +955,644 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "53":
+    id: "53"
+    taskid: 6913117a-5072-45c9-81ac-789a2b1e84ec
+    type: regular
+    task:
+      id: 6913117a-5072-45c9-81ac-789a2b1e84ec
+      version: -1
+      name: Set ASM enrichment status field (AWS)
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_AWS_Enrichment.yml:1473]:
```suggestion
      name: Set ASM enrichment status to false (AWS)
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Qualys_Enrichment.yml [hunk 9]
```diff
@@ -612,14 +612,104 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "17":
+    id: "17"
+    taskid: c737255d-4571-477a-8887-b29ea923efc1
+    type: regular
+    task:
+      id: c737255d-4571-477a-8887-b29ea923efc1
+      version: -1
+      name: Set ASM - Enrichment Status
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Qualys_Enrichment.yml:639]:
```suggestion
      name: Set ASM enrichment status to true
```
Consistency with other playbooks

File: Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Qualys_Enrichment.yml [hunk 10]
```diff
@@ -612,14 +612,104 @@ tasks:
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "19":
+    id: "19"
+    taskid: ea5e1f1e-027e-47ab-8765-0eec317445d0
+    type: regular
+    task:
+      id: ea5e1f1e-027e-47ab-8765-0eec317445d0
+      version: -1
+      name: Set ASM - Enrichment Status
```
Comment (johnnywilkes) [inline Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Qualys_Enrichment.yml:684]:
```suggestion
      name: Set ASM enrichment status to false
```
Consistency with other playbooks

**Review by johnnywilkes — CHANGES_REQUESTED** [review]
some nits for consistency sakes

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
+#### Incident Fields
+
+##### ASM - Enrichment Status
+
+Added a new field, **ASM - Enrichment Status**, to track the status of types of platforms enrichment has been performed with.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md:5]:
```suggestion
Added a new field, **ASM - Enrichment Status**, to track the status of types of platform enrichment that has been performed.
```

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+##### ASM - Dev Check Details
+
+Update the comment field of the `result` key.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md:9]:
```suggestion
Updated the comment field of the *result* key.
```

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+Playbook now supports populating the ASM - Enrichment Status field to identify if enrichment was completed successfully or not.
+
+##### Cortex ASM - Azure Enrichment
+
+- Playbook now supports populating the ASM - Enrichment Status field to identify if enrichment was completed successfully or not.
+- Updated the use of `N/A` to `n/a` for when various results are not found in the playbook to follow internal conventions.
+
+##### Cortex ASM - Tenable.io Enrichment
+
+- Playbook now supports populating the ASM - Enrichment Status field to identify if enrichment was completed successfully or not.
```
Comment (ShirleyDenkberg) [inline Packs/CortexAttackSurfaceManagement/ReleaseNotes/1_6_27.md:39]:
```suggestion
Playbook now supports populating the ASM - Enrichment Status field to identify if enrichment was completed successfully or not.
```

**Comment by ShirleyDenkberg** [comment]
@johnnywilkes @anas-yousef @efelmandar Doc review completed.

**Comment by BigEasyJ** [comment]
@anas-yousef this PR is ready for merge.

**Review by anas-yousef — APPROVED** [review]

**Pull request merged**